app/utils/file/file_picker/index.ts

```diff
@@ -0,0 +1,331 @@
+                }),
+            },
+            denied_android: {
+                title: formatMessage(
+                    {
+                        id: 'mobile.android.photos_permission_denied_title',
+                        defaultMessage:
+                            '{applicationName} would like to access your photos',
+                    },
+                    {applicationName},
+                ),
+                text: formatMessage({
+                    id: 'mobile.android.photos_permission_denied_description',
+                    defaultMessage:
+                        'Upload photos to your Mattermost instance or save them to your device. Open Settings to grant Mattermost Read and Write access to your photo library.',
+                }),
+            },
+        };
+
+        return permissions[source];
+    };
+
+    private prepareFileUpload = async (files: Array<Asset | DocumentPickerResponse>) => {
+        const out = await extractFileInfos(files);
+
+        if (out.length > 0) {
+            DeviceEventEmitter.emit(Navigation.NAVIGATION_CLOSE_MODAL);
+            this.uploadFiles(out);
+        }
+    };
+
+    private getPermissionDeniedMessage = (source?: PermissionSource) => {
+        const sources = ['camera', 'storage', 'photo'];
+        const deniedSource: PermissionSource = Platform.select({android: 'denied_android', ios: 'denied_ios'})!;
+        const msgForSource = source && sources.includes(source) ? source : deniedSource;
+
+        return this.getPermissionMessages(msgForSource);
+    };
+
+    private getFilesFromResponse = async (response: ImagePickerResponse): Promise<Asset[]> => {
+        if (!response?.assets?.length) {
+            return [];
+        }
+
+        const files: Asset[] = [];
+
+        await Promise.all((response.assets.map(async (file) => {
+            if (Platform.OS === 'android') {
+                // For android we need to retrieve the realPath in case the file being imported is from the cloud
+                const uri = (await ShareExtension.getFilePath(file.uri)).filePath;
+                const type = file.type || lookupMimeType(uri);
+                let fileName = file.fileName;
+                if (type.includes('video/')) {
+                    fileName = uri.split('\\').pop().split('/').pop();
+                }
+
+                if (uri) {
+                    files.push({...file, fileName, uri, type});
+                }
+            } else {
+                files.push(file);
+            }
+        })));
+
+        return files;
+    };
+
+    private hasPhotoPermission = async (source: PermissionSource) => {
+        let permissionRequest;
+        let targetSource: AndroidPermission | IOSPermission = source === 'camera' ? Permissions.PERMISSIONS.IOS.CAMERA : Permissions.PERMISSIONS.IOS.PHOTO_LIBRARY;
```

Comment:
Why not use platform select? It is practically the same, but might be a bit more readable.

Reply:
Done